Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/Detect_features_in_polyhedra.h

```diff
@@ -0,0 +1,329 @@
+       (angle_in_deg != FT(180) && is_sharp(polyhedron,he,cos_angle))
+       )
+    {
+      put(hif, he, true);
+      put(hif, opposite(he,polyhedron), true);
+      
+      put(vnfe, target(he,polyhedron), get(vnfe, target(he,polyhedron))+1);
+      put(vnfe, source(he,polyhedron), get(vnfe, source(he,polyhedron))+1);
+    }
+  }
+}
+
+
+template <typename P_, typename I_>
+template <typename Int>
+Int
+Detect_features_in_polyhedra<P_, I_>::
+generate_patch_id(Int, int i)
+{
+  return Int(i);
+}
+
+template <typename P_, typename I_>
+template <typename Int>
+std::pair<Int, Int>
+Detect_features_in_polyhedra<P_, I_>::
+generate_patch_id(std::pair<Int, Int>, int i)
+{
+  return std::pair<Int, Int>(i, 0);
+}
+
+template <typename P_, typename I_>
+void
+Detect_features_in_polyhedra<P_, I_>::
+detect_surface_patches(Polyhedron& polyhedron)
+{
+  // Initialize unsorted_faces
+  face_descriptor_set unsorted_faces;
+  for ( typename boost::graph_traits<Polyhedron>::face_iterator fit = faces(polyhedron).begin(),
+       end = faces(polyhedron).end() ; fit != end ; ++fit )
+  {
+
+    unsorted_faces.insert(*fit);
+  }
+  
+  // Flood
+  while ( ! unsorted_faces.empty() )
+  {
+    face_descriptor f = *(unsorted_faces.begin());
+    unsorted_faces.erase(unsorted_faces.begin());
+    
+    const Patch_id patch_id = generate_patch_id(Patch_id(),
+                                                current_surface_index_);
+    put(pid_map, f, patch_id);
+    flood(polyhedron, f,patch_id,unsorted_faces);
+    ++current_surface_index_;
+  }
+}
+
+
+template <typename P_, typename I_>
+void
+Detect_features_in_polyhedra<P_, I_>::
+detect_vertices_incident_patches(Polyhedron& polyhedron)
+{
+  typename boost::property_map<Polyhedron,halfedge_is_feature_t>::type hif
+      = get(halfedge_is_feature,polyhedron);
+  typedef typename boost::property_map<Polyhedron,vertex_incident_patches_t>::type VIP_map;
+   VIP_map vip = get(vertex_incident_patches,polyhedron);
+
+  BOOST_FOREACH(vertex_descriptor vit,vertices(polyhedron))
+  {
+    // Look only at feature vertices
+    if( ! get(hif, halfedge(vit, polyhedron)) ){ continue; }
+    
+    // Loop on incident facets of vit
+    std::set<Patch_id> set;
+    BOOST_FOREACH(halfedge_descriptor he, halfedges_around_target(vit,polyhedron))
+    {
+      if( ! is_border(he,polyhedron) )
+      {
+        set.insert(get(pid_map,face(he,polyhedron)));
+      }
+      else if( ! is_border(opposite(he,polyhedron),polyhedron) )
+      {
+        set.insert(get(pid_map, face(opposite(he,polyhedron),polyhedron)));
+      }
+    }
+    put(vip, vit, set);
+  }
+}
+  
+// -----------------------------------
+// Private methods
+// -----------------------------------
+template <typename P_, typename I_>
+typename Detect_features_in_polyhedra<P_, I_>::Vector_3
+Detect_features_in_polyhedra<P_, I_>::
+facet_normal(const Polyhedron& polyhedron, const face_descriptor& f) const
+{
```

Comment:
you can replace this function by a simple call to `PMP::compute_facet_normal` everywhere in the file